rclcpp/include/rclcpp/executor.hpp

```diff
@@ -305,7 +305,9 @@ class Executor
    * If the time that waitables take to be executed is longer than the period on which new waitables
    * become ready, this method will execute work repeatedly until `max_duration` has elapsed.
    *
-   * \param[in] max_duration The maximum amount of time to spend executing work. Must be positive.
+   * \param[in] max_duration The maximum amount of time to spend executing work, must be >= 0.
+   *   `0` means it keeps executing until no more work is available.
```

Comment:
It should be made clear in the docs that when passing timeout=0, the method could potentially block forever (if more work to be done is found after executing the previous ready executables)

Reply:
> if more work to be done is found after executing the previous ready executables

I see. if this situation continues, potentially block forever, since there is no timeout.

Reply:
fixed in https://github.com/ros2/rclcpp/pull/1878/commits/4dd2b772cf8373fd7bf5032172fa3d59de652ee2